### DISCUSSION TITLE
Error when trying to run demo_multisession script

### DISCUSSION THREAD

**Discussion by bioBaddie** [flatironinstitute/CaImAn#1232] [Q&A] [answered]
Hi everyone! I am attempting to run the demo script for multisession registration, but ran into this error:

DeprecationWarning: Please use `csc_matrix` from the `scipy.sparse` namespace, the `scipy.sparse.csc` namespace is deprecated.
  data = pickle.load(infile)

I am still fairly new to Python, so I wasn't sure how I should update to prevent to not rely on this deprecated function? 

Thank you!!

**Comment by EricThomson** [comment]
Thanks for the question, and sorry for the delay in response.

This is something we should update. I'll convert this to an issue thanks for bringing it to our attention!

Just to be clear you are getting a warning, not an error that stops the code from running?

Opened an issue: 
https://github.com/flatironinstitute/CaImAn/issues/1235

**Reply by bioBaddie** [reply]
Thank you for looking into this!

I haven't tried using my own data on this script - I just tried running the code as is with the data provided. The error wouldn't allow me to run any subsequent cells. Any suggestions? 

I was concatenating multiple videos (in vitro recordings) together and then motion correcting the concatenated video before CNMF analysis. I know this isn't the best approach so I was hoping to figure out how to use the multi-session function to track the same neurons across multiple recordings. 

Thanks again! :)

**Reply by EricThomson** [reply]
Oh so this wasn't an innocuous warning it was actually stopping from running. Could you go to the issue and show the full stack trace?

And what version of Caiman you are running?

**Reply by pgunn** [reply, marked as answer]
@AB0913 Hello, I fixed the sample dataset to handle the upstream changes in scipy. If you remove the alignment.pickle sample dataset from your caiman_data directory, it will be refreshed the next time you run the notebook and you won't see that warning anymore.

(if the rest of the notebook was not running, it was probably not running for some other reason and we'll be happy to help you figure out why if you can get us the messages you got on the cell after this one)

**Reply by bioBaddie** [reply]
Thank you for fixing the error - the code runs great now! I did have one other question relating to the multi-session registration - I saved my cnmfe results to and hdf5 using the built in save function. I can easily find the 'A' and 'dims' that are needed as inputs for the multi-session registration, but am not sure where the template image is being saved / if I need to manually do this? Thank you again for the help! :)

**Reply by EricThomson** [reply]
You can sneak it into the estimates object by adding `cnmf_object.estimates.corr_image = corr_image` and then it will save/load with everything else.
